core/src/main/java/hudson/model/Queue.java

```diff
@@ -1460,6 +1487,9 @@ public void maintain() {
             // allocate buildable jobs to executors
             for (BuildableItem p : new ArrayList<BuildableItem>(
                     buildables)) {// copy as we'll mutate the list in the loop
+                if (p.task instanceof FlyweightTask){
```

Comment:
Where do we allocate tasks to executors? Seems features like Load balancers or build blockers will never work for flyweight tasks after the change